### ISSUE TITLE
Add support for document revision histories

### ISSUE THREAD

**Issue by david-waltermire-nist** [usnistgov/OSCAL#587]
# User Story:

As an OSCAL content creator, I need a way to provide a document revision history.

## Goals:

Add support for a revision history in the document metadata.

## Dependencies:

None.

## Acceptance Criteria

- [ ] All [OSCAL website](https://pages.nist.gov/OSCAL) and readme documentation affected by the changes in this issue have been updated. Changes to the OSCAL website can be made in the docs/content directory of your branch.
- [ ] A Pull Request (PR) is submitted that fully addresses the goals of this User Story. This issue is referenced in the PR.
- [ ] The CI-CD build process runs without any reported errors on the PR. This can be confirmed by reviewing that all checks have passed in the PR.

{The items above are general acceptance criteria for all User Stories. Please describe anything else that must be completed for this issue to be considered resolved.}

**Comment by david-waltermire-nist** [comment]
The following are notes from a previous discussion.

Current state:

```
metadata:
  published: date : date the OSCAL instance was published
  last-modified: date : date the OSCAL instance was last modified
  version: string : version of the OSCAL instance
```

Future state:

```
metadata:
  revision-history: revision history of the OSCAL document based on feedback from #498
  published: date : date the OSCAL instance was published
  last-modified: date : date the OSCAL instance was last modified
  version: string : version of the OSCAL instance
  link: rel="source" href="#pointer": pointer to a resource indicating the source the OSCAL instance was derived from
back-matter:
  resources/citations:
    resource/citation:
      id: used for the source pointer above
      properties:
        prop:
          name: published
          value: date
```

- merge citation into resource
  - add title and document-ids
  - add "accessed-on: date"
  - review FedRAMP extensions

**Comment by bradh** [comment]
+1 on revision history as a need.

Side note: `rel="source"` doesn't appear in https://www.iana.org/assignments/link-relations/link-relations.xhtml. I thought perhaps `derivedFrom`.

**Comment by brianrufgsa** [comment]
Added separate issue  #612 for the merging of citation into resource.

**Comment by david-waltermire-nist** [comment]
This issue was also referenced in #498. Here is the specific design in pseudo-YAML from that issue:
```
versions:
  version:
    published: 
    last-modified: 
    version: 
    oscal-version: 1.0
    uri: over links for uniqueid href and integrity
```

**Comment by brianrufgsa** [comment]
# 30-Jan-2020 Status
@david-waltermire-nist implemented changes in the metaschema to support this. Those changes are in his branch and require a PR.

**Comment by david-waltermire-nist** [comment]
This is addressed by PR #614.
